subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/PrecompiledScriptPluginIntegrationTest.kt

```diff
@@ -211,4 +211,94 @@ class PrecompiledScriptPluginIntegrationTest : AbstractPluginIntegrationTest() {
     }
+
+    @Test
+    @ToBeFixedForInstantExecution(because = "Kotlin Gradle Plugin")
+    fun `accessors are available after renaming precompiled script plugin from project dependency`() {
+
+        requireGradleDistributionOnEmbeddedExecuter()
+
+        withFolders {
+            withFile("settings.gradle.kts", """
+                $defaultSettingsScript
+
+                include("consumer", "producer")
+            """)
+
+            withFile("build.gradle.kts", """
+                plugins {
+                    `java-library`
+                    `kotlin-dsl` apply false
+                }
+
+                allprojects {
+                    $repositoriesBlock
+                }
+
+                dependencies {
+                    api(project(":consumer"))
+                }
+            """)
+
+            "consumer" {
+                withFile("build.gradle.kts", """
+                    plugins {
+                        id("org.gradle.kotlin.kotlin-dsl")
+                    }
+
+                    configurations {
+                        compileClasspath {
+                            attributes {
+                                // Forces dependencies to be visible as jars
+                                // to reproduce the failure that happens in forkingIntegTest.
+                                // Incidentally, this also allows us to write `stable-producer-plugin`
+                                // in the plugins block below instead of id("stable-producer-plugin").
+                                attribute(
+                                    LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE,
+                                    objects.named(LibraryElements.JAR)
+                                )
+                            }
+                        }
+                    }
+
+                    dependencies {
+                        implementation(project(":producer"))
+                    }
+                """)
+
+                withFile("src/main/kotlin/consumer-plugin.gradle.kts", """
+                    plugins { `stable-producer-plugin` }
+                """)
+            }
+
+            "producer" {
+                withFile("build.gradle.kts", """
+                    plugins { id("org.gradle.kotlin.kotlin-dsl") }
+                """)
+                withFile("src/main/kotlin/changing-producer-plugin.gradle.kts")
+                withFile("src/main/kotlin/stable-producer-plugin.gradle.kts", """
+                    println("*42*")
+                """)
+            }
+        }
+
+        build("assemble").run {
+            assertTaskExecuted(
+                ":consumer:generateExternalPluginSpecBuilders"
+            )
+        }
+
+        existing("producer/src/main/kotlin/changing-producer-plugin.gradle.kts").run {
+            java.nio.file.Files.move(
+                toPath(),
+                resolveSibling("changed-producer-plugin.gradle.kts").toPath()
+            )
+        }
```

Comment:
💄 can be simplified to
```kotlin
existing("producer/src/main/kotlin").run {
    resolve("changing-producer-plugin.gradle.kts").renameTo(resolve("changed-producer-plugin.gradle.kts"))
}
```

Reply:
That was actually my 1st version but I'm trying to avoid `renameTo` after seeing problems on windows.

Reply:
Sounds weird because it's the same directory, same filesystem.

Reply:
I've changed the code to use `renameTo` for the improved prose.